src/Domain/BlockLogicalCoordinates.cpp

```diff
@@ -39,24 +42,69 @@ std::vector<block_logical_coord_holder<Dim>> block_logical_coordinates(
     // the smallest block_id).
     for (const auto& block : domain.blocks()) {
       if (block.is_time_dependent()) {
-        const auto moving_inv =
-            block.moving_mesh_grid_to_inertial_map().inverse(x_frame);
-        if (not moving_inv) {
-          continue;
-        }
-        const auto inv =
-            block.moving_mesh_logical_to_grid_map().inverse(moving_inv.get());
-        if (inv) {
-          x_logical = inv.get();
-        } else {
-          continue;  // Not in this block
+        if constexpr (std::is_same_v<Frame, ::Frame::Inertial>) {
+          // Point is in the inertial frame, so we need to map to the grid
+          // frame and then the logical frame.
+          const auto moving_inv =
+              block.moving_mesh_grid_to_inertial_map().inverse(
+                  x_frame, time, functions_of_time);
+          if (not moving_inv) {
+            continue;
+          }
+          // logical to grid map is time-independent.
+          const auto inv =
+              block.moving_mesh_logical_to_grid_map().inverse(moving_inv.get());
+          if (inv) {
+            x_logical = inv.get();
+          } else {
+            continue;  // Not in this block
+          }
+        } else {  // frame is different than ::Frame::Inertial
+          // Currently 'time' is unused in this branch.
+          // To make the compiler happy, need to trick it to think that
+          // 'time' is used.
+          (void) time;
+          // Currently the only map that a Block can return is GridToInertial.
```

Comment:
Currently we only support Grid and Inertial frames in the block, so make sure Frame is ::Frame::Grid. ( The Inertial case was handled above.)